armi/utils/tests/test_utils.py

```diff
@@ -161,6 +192,84 @@ def test_list2str(self):
+
+    def test_relErr(self):
+        self.assertAlmostEqual(utils.relErr(1.00, 1.01), 0.01)
+        self.assertAlmostEqual(utils.relErr(100.0, 97.0), -0.03)
+        self.assertAlmostEqual(utils.relErr(0.00, 1.00), -1e99)
+
+    def test_efmt(self):
+        self.assertAlmostEqual(utils.efmt("1.0e+001"), "1.0E+01")
+        self.assertAlmostEqual(utils.efmt("1.0E+01"), "1.0E+01")
+
+    def test_slantSplit(self):
+        x1 = utils.slantSplit(10.0, 4.0, 4)
+        x2 = utils.slantSplit(10.0, 4.0, 4, order="high first")
+        self.assertListEqual(x1, [1.0, 2.0, 3.0, 4.0])
+        self.assertListEqual(x2, [4.0, 3.0, 2.0, 1.0])
+
+    def test_newtonsMethod(self):
+        f = lambda x: (x + 2) * (x - 1)
+        root = utils.newtonsMethod(f, 0.0, 5.0, maxIterations=10, positiveGuesses=True)
+        self.assertAlmostEqual(root, 1.0, places=3)
+        root = utils.newtonsMethod(f, 0.0, -10.0, maxIterations=10)
+        self.assertAlmostEqual(root, -2.0, places=3)
+
+    def test_minimizeScalarFunc(self):
+        f = lambda x: (x + 1) ** 2
+        minimum = utils.minimizeScalarFunc(f, -3.0, 10.0, maxIterations=10)
+        self.assertAlmostEqual(minimum, -1.0, places=3)
+        minimum = utils.minimizeScalarFunc(
+            f, -3.0, 10.0, maxIterations=10, positiveGuesses=True
+        )
+        self.assertAlmostEqual(minimum, 0.0, places=3)
+
+    def test_prependToList(self):
+        a = ["hello", "world"]
+        b = [1, 2, 3]
+        utils.prependToList(a, b)
+        self.assertListEqual(a, [1, 2, 3, "hello", "world"])
+
+    def test_convertToSlice(self):
+        slice1 = utils.convertToSlice(2)
+        self.assertEqual(slice1, slice(2, 3, None))
+        slice1 = utils.convertToSlice(2.0, increment=-1)
+        self.assertEqual(slice1, slice(1, 2, None))
+        slice1 = utils.convertToSlice(None)
+        self.assertEqual(slice1, slice(None, None, None))
+        slice1 = utils.convertToSlice([1, 2, 3])
+        self.assertTrue(np.allclose(slice1, np.array([1, 2, 3])))
+        slice1 = utils.convertToSlice(slice(2, 3, None))
+        self.assertEqual(slice1, slice(2, 3, None))
+        slice1 = utils.convertToSlice(np.array([1, 2, 3]))
+        self.assertTrue(np.allclose(slice1, np.array([1, 2, 3])))
+        with self.assertRaises(Exception):
+            slice1 = utils.convertToSlice("slice")
+
+    def test_plotMatrix(self):
+        matrix = np.zeros([2, 2], dtype=float)
+        matrix[0, 0] = 1
+        matrix[0, 1] = 2
+        matrix[1, 0] = 3
+        matrix[1, 1] = 4
+        xtick = ([0, 1], ["1", "2"])
+        ytick = ([0, 1], ["1", "2"])
+        dc = directoryChangers.TemporaryDirectoryChanger()
+        dc.__enter__()
+        fname = "test_plotMatrix_testfile"
+        utils.plotMatrix(matrix, fname, show=True, title="plot")
+        utils.plotMatrix(matrix, fname, minV=0, maxV=5, figsize=[3, 4])
+        utils.plotMatrix(matrix, fname, xticks=xtick, yticks=ytick)
+        dc.__exit__(None, None, None)
```

Comment:
is this where that os.remove() used to be? the DirectoryChangers are actually [context managers](https://docs.python.org/3/reference/datamodel.html#context-managers), meaning that you can enter them with a `with ...:` statement. Like:
```
with directoryChangers.TemporaryDirectoryChanger(): # __enter__() called implicitly here
    # do stuff
# __exit__() implicitly called here
```

Reply:
got it. my initial attempt at this type of usage didn't work how I wanted it, I think because I was using the wrong method. I copied that usage from a setUp/tearDown in a unit test, where enter/exit are called explicitly because they're in different functions. I'll fix this.